[Official release] Multicurrency variable config + new dbt_current.timestamp() configuration (#22)
fivetran/dbt_zuora: CHANGELOG.md

# dbt_zuora v0.3.2
This release introduces the following updates.

## Bug Fixes  

> - Flipped the variable logic in `zuora__using_multicurrency` in these models so that the `*_amount` values are called by default rather than `*_amount_home_currency` when the variable isn't set: ([#22](https://github.com/fivetran/dbt_zuora/pull/22))

Reviewer: ```suggestion
- Resolved flipped logic around the `zuora__using_multicurrency` [variable](https://github.com/fivetran/dbt_zuora?tab=readme-ov-file#step-5-configure-the-multicurrency-variable-for-customers-billing-in-multiple-currencies). ([#22](https://github.com/fivetran/dbt_zuora/pull/22))
  - Ensured that, for multicurrency users, `*_amount_home_currency` fields are used and that `*_amount` values are used for single-currency users. Previously, this was erroneously flipped.
```
Just making it more readable but please adjust if this is incorrect in any way!

Reviewer: same for pre-release notes

Author: Looks way better!

  - `zuora__line_item_history`: For setting the `gross_revenue` and `discount_revenue` values (these values are then used to calculate `net_revenue`).
  - `zuora__monthly_recurring_revenue`: For setting the `mrr_expected_current_month` value.

>   - `int_zuora__transaction_grouped`: For setting daily amount values for invoices, discounts, taxes and credit balance adjustments to flow downsteram into the `zuora__account_daily_overview` model.

Reviewer: This part is a lil confusing -- are these the affected fields for each end model? If so, I would phrase it more like "This affects..." rather than "For setting..."

Reviewer: same for pre-release

Author: Updated.

> - Multicurrency customers must set the `zuora__using_multicurrency` variable to `true` to enable that functionality. ([#22](https://github.com/fivetran/dbt_zuora/pull/22))

Reviewer: What do you mean by "that functionality"?

Reviewer: Also if this relates to the above changes, I'd say it should be a sub-bullet or 
> a note like this

- Leveraged the `{{ dbt.type_timestamp() }}` macro within staging models for all timestamp fields. Certain Redshift warehouses sync these fields as `timestamp with time zone` fields by default, causing errors in the `zuora` package. This macro appropriately removes timezone values from the UTC timestamps and ensures successful compilations of these models. 
  - For more details, please refer to the [dbt_zuora_source v0.2.2 release](https://github.com/fivetran/dbt_zuora_source/releases/tag/v0.2.2).

> - Updated `subscription_period_started_at` and `subscription_period_ended_at` with the `{{ dbt.type_timestamp() }}` cast in `zuora__line_item_enhanced` to remove date/timestamp mismatches on the `union all` function. ([#20](https://github.com/fivetran/dbt_zuora/pull/20))

Reviewer: ```suggestion
- Also cast `subscription_period_started_at` and `subscription_period_ended_at` as `{{ dbt.type_timestamp() }}` in `zuora__line_item_enhanced` to remove date/timestamp mismatches on the `union all` function. ([#20](https://github.com/fivetran/dbt_zuora/pull/20))
```

Reviewer: same for the pre-release notes

Author: Updated.

## Under the Hood
- Replaced the deprecated `dbt.current_timestamp_backcompat()` function with `dbt.current_timestamp()` to ensure all timestamps are captured in UTC.  ([#20](https://github.com/fivetran/dbt_zuora/pull/20))

> - This change is applied in the following end models:
>   - `zuora__billing_history`
>   - `zuora__line_item_history`
>   - `zuora__subscription_overview`
> - As well as the intermediate models:
>   - `int_zuora__account_enriched`
>   - `int_zuora__mrr_date_spine`
>   - `int_zuora__transaction_date_spine`

Reviewer: ```suggestion
  - This change is applied in the following end models:
    - `zuora__billing_history`
    - `zuora__line_item_history`
    - `zuora__subscription_overview`
  - As well as the intermediate models:
    - `int_zuora__account_enriched`
    - `int_zuora__mrr_date_spine`
    - `int_zuora__transaction_date_spine`
```
nest it under the first bullet

Reviewer: should apply to the pre-release notes as well

Author: Committed.
